libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1422,7 +1422,25 @@ bool TypeChecker::visit(Conditional const& _conditional)
 		BOOST_THROW_EXCEPTION(FatalError());
 	else if (trueType && falseType)
 	{
-		commonType = Type::commonType(trueType, falseType);
+		if (trueType->category() == Type::Category::InlineArray &&
+			falseType->category() == Type::Category::InlineArray)
+		{
+			auto const* inlineArrayTrueType = dynamic_cast<InlineArrayType const*>(trueType);
```

Comment:
Please always use `dynamic_cast<InlineArrayType const&>` unless you are not sure of the type.